test/gc.wast.fromBinary

```diff
@@ -0,0 +1,150 @@
+(module
+ (type $none_=>_none (func))
+ (import "env" "trivial_i31ref" (global $gimport$0 i31ref))
+ (global $global$0 (mut anyref) (ref.null any))
+ (global $global$1 (mut eqref) (ref.null eq))
+ (global $global$2 (mut i31ref) (global.get $gimport$0))
+ (global $global$3 (mut anyref) (ref.null eq))
+ (global $global$4 (mut anyref) (global.get $gimport$0))
+ (global $global$5 (mut eqref) (global.get $gimport$0))
+ (func $test
+  (local $local_anyref i32)
+  (local $local_eqref anyref)
+  (local $local_i31ref eqref)
+  (local $local_i32 i31ref)
```

Comment:
Namings in the roundtripped binary are off, but the code is otherwise correct. Wondering what's causing this.

Reply:
Looks like the problem consists of two parts:
* Locals are reordered, magically moving the i32 local from index 3 in the original to index 0 after round-tripping
* Names are not updated accordingly

New ordering seems to adhere to declaration order in `Type::BasicID`. Might make sense to postpone adding the test so this can be investigated?

Reply:
Alternative: Just moved the local in the test, so the error does not manifest and we can look into it in a follow-up.